sale_order_type/models/sale.py

```diff
@@ -33,6 +33,19 @@ def _default_type_id(self):
             [("company_id", "in", [company_id, False])], limit=1
         )
 
+    @api.model
+    def _default_sequence_id(self):
+        """We get the sequence in same way the core next_by_code method does so we can
+        get the proper default sequence"""
+        force_company = self.env.context.get("force_company")
+        if not force_company:
+            force_company = self.company_id.id or self.env.company.id
+        return self.env["ir.sequence"].search(
+            [("code", "=", "sale.order"), ("company_id", "in", [force_company, False])],
+            order="company_id",
```

Comment:
why order="company_id" ?

Reply:
Is the same order `next_by_code()` does it: https://github.com/OCA/OCB/blob/13.0/odoo/addons/base/models/ir_sequence.py#L280